### PR TITLE
Issue #3260572: Filter groups on access on all-groups page.

### PR DESCRIPTION
**Pull request by NickDaelemans**
## Problem
When you look at the 'All Groups' overview (/all-groups) you can also see community groups. The view is not filters sufficiently on group access.

## Solution
Adding a ViewsFilter and applying it using config override class should solve this issue.

## Issue tracker
https://getopensocial.atlassian.net/browse/SUP-2034
https://www.drupal.org/project/social/issues/3260572

## How to test
- [ ] Create a community group (open_group or flexible with right visibility)
- [ ] Go to /all-groups as anonymous user
- [ ] You are not able to see that new group (since you don't have access to it)
- [ ] Feel free to also check this with closed_group or flexible (member groups).

## Release notes
Groups are now filtered on right access when viewing the all-groups page. You will not be able to see closed or community groups when being anonymous or not in the right group.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2713-prt6uyamyeta9g0k3nnlpyil81liuiaj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f1280e046a4b8d291e9d8f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2713-prt6uyamyeta9g0k3nnlpyil81liuiaj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f1280e046a4b8d291e9d8f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2713-prt6uyamyeta9g0k3nnlpyil81liuiaj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f1280e046a4b8d291e9d8f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2713-prt6uyamyeta9g0k3nnlpyil81liuiaj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f1280e046a4b8d291e9d8f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2713-prt6uyamyeta9g0k3nnlpyil81liuiaj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f1280e046a4b8d291e9d8f

**Comment by nkoporec** [comment]
@NickDaelemans I could reproduce the issue with flexible groups and visibility set to community. After applying your patch the issue is fixed, however, I think it would be better to combine your changes with the `social_group_flexible_group_views_query_alter`.

This hook is responsible for the access control for the /all-groups page, so it would be better to have everything in the same place.

**Comment by Ressinel** [comment]
@NickDaelemans We should avoid using **ConfigOverride**. Instead create `hook_update_N()`.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2713-prt6uyamyeta9g0k3nnlpyil81liuiaj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f1280e046a4b8d291e9d8f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2713-prt6uyamyeta9g0k3nnlpyil81liuiaj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f1280e046a4b8d291e9d8f

File: modules/social_features/social_group/social_group.module [hunk 1]
```diff
@@ -2107,63 +2140,32 @@ function social_group_get_allowed_visibility_options_per_group_type($group_type_
...
   }
-
-  $found = FALSE;
-
-  foreach ($query->where as &$conditions) {
-    foreach ($conditions['conditions'] as &$condition) {
-      if ($condition['field'] == 'groups_field_data.type') {
-        $found = TRUE;
-        break 2;
+  $account = \Drupal::currentUser()->getAccount();
```
Comment (NickDaelemans) [inline modules/social_features/social_group/social_group.module:2146]: This is the important part :)

**Comment by NickDaelemans** [comment]
@nkoporec I updated this PR. I altered the social_group_views_query_alter function. Not sure why, but the entire logic inside that hook could be removed and I added a check for the closed_group only.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2713-prt6uyamyeta9g0k3nnlpyil81liuiaj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f1280e046a4b8d291e9d8f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2713-prt6uyamyeta9g0k3nnlpyil81liuiaj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f1280e046a4b8d291e9d8f

**Review by nkoporec — APPROVED** [review]
.

**Pull request merged**